### PR TITLE
[JENKINS-62202] Fix accidental read-only mode in jobs and folders

### PR DESCRIPTION
**Pull request by daniel-beck**
…gurations

----

[JENKINS-62202](https://issues.jenkins-ci.org/browse/JENKINS-62202)

Regression because of https://github.com/jenkinsci/matrix-auth-plugin/pull/78

### Testing notes

Enable permissions involved like this:

```
Item.EXTENDED_READ.enabled = true
Jenkins.SYSTEM_READ.enabled = true
Computer.EXTENDED_READ.enabled = true
```

`mvn hpi:run -Djenkins.version=2.235` helps some, but Computer/Configure can only be confirmed with https://github.com/jenkinsci/jenkins/pull/4531.

Manual testing setup:

1. `docker run --rm -ti -p 8080:8080 -e ID=4531 jenkins/core-pr-tester`
2. In Setup Wizard, install _only_ Folders Plugin
3. Create admin user
4. In Plugin Manager » Advanced, upload PR build
5. Create Manage Users, create user/user
6. In Script Console, run the script above

Now permissions can be assigned at different levels to ensure this works as intended.

### CODE REVIEW

File: src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly [hunk 1]
```diff
@@ -98,7 +98,9 @@ THE SOFTWARE.
             ${g.title}
           </td>
         </j:forEach>
-        <td rowspan="2" class="stop" />
+        <l:hasPermission permission="${instance.getEditingPermission()}">
+          <td rowspan="2" class="stop" />
+        </l:hasPermission>
```
Comment (daniel-beck) [inline src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly:103]: This fixes a visual artifact caused by the previous PR.

File: src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly [hunk 2]
```diff
@@ -60,7 +60,7 @@ THE SOFTWARE.
             </j:if>
           </j:forEach>
         </j:forEach>
-        <l:isAdmin>
+        <l:hasPermission permission="${instance.getEditingPermission()}">
```
Comment (timja) [inline src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly:63]: I think you could use instance.editingPermission?

Comment (daniel-beck) [reply src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly:63]: 🤷 no difference there, and makes grepping slightly more straightforward.

**Comment by timja** [comment]
Computer.EXTENDED_READ:

image

**Comment by daniel-beck** [comment]
This doesn't work when viewing cloud agent templates for users with Computer/Configure and Overall/SystemRead but lacking Overall/Administer:

> Screenshot

Probably a small price to pay to get this fix, but still 😫

**Pull request merged**